ci/upload-console.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+source ci/upload-ci-artifact.sh
+
+set -e
+
+if [[ -n $CI ]]; then
+  exit_code=0
+  console_out="console-$(date '+%Yy%mm%dd%Hh%Mm%Ss%Nns').log"
+
+  # shellcheck disable=SC2094
+  if "$@" 2>> "$console_out" 1>> >(tee -a "$console_out"); then
+    # noop
+    true
+  else
+    exit_code=$?
+  fi
+  gzip -f "$console_out"
+  upload-ci-artifact "$console_out.gz"
```

Comment:
maybe this solution's downside would be no upload form this line if the build is manually cancelled or timed out.

Maybe, add some hidden ENV or see RETRY_COUNT to full output mode?

Reply:
we can put things into post-command hook. like this https://github.com/solana-labs/solana/blob/master/.buildkite/hooks/post-command#L23-L30 so that it will always upload what you want 👻